examples: ensure the proper build order

jbech-linaro opened the pull request:
The Makefiles for the examples lacked dependencies between rules, resulting in some cases resulted in build error. To maintain a coherent build process, this commit introduces dependencies to ensure that binaries are built before being stripped and signed.

b49020 commented:
`Reviewed-by: Sumit Garg <sumit.garg@linaro.org>`

@DemesneGH Can you trigger the CI here?

jbech-linaro commented:
R-B tag added and fixed a repetition in the commit message.

DemesneGH commented:
`Acked-by: Yuan Zhuang <yuanz@apache.org>` 
thanks!

jbech-linaro commented:
Thanks @DemesneGH ! Looked like the CI passed, I've added your Acked-by tag and force pushed.

Squashed, rebased and tag(s) applied (should be) ready for merge! Thanks!

jbech-linaro commented:
Thanks @b49020 and @DemesneGH for the quick turn around time on this one! Much appreciated!
